### PR TITLE
Implement DJTAG2 protocol, add fast SPI

### PR DESCRIPTION
**Pull request by phdussud**
After quite a bit of testing I would like to submit my support for DJTAG2 and higher programming speed.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,4 +4,8 @@ src/*.d
 src/*.o
 src/*.map
 *.bin
-*.elf
\ No newline at end of file
+*.elf
+/.cproject
+/.project
+/.vscode/
+/dirtyjtag Default.launch
```
Comment (jeanthom) [inline .gitignore]: IDE-specific files shouldn't be added to DirtyJTAG's .gitignore. I recommend you to setup a global gitignore file for your system: https://gist.github.com/subfuzion/db7f57fff2fb6998a16c ;-)

File: Makefile [hunk 2]
```diff
@@ -18,6 +18,7 @@ CFLAGS += -Wall -Wextra -Werror
 CFLAGS += -fno-common -ffunction-sections -fdata-sections
 CFLAGS += -std=gnu11
 CFLAGS += -DPLATFORM='HW_$(PLATFORM)'
+#CFLAGS += -Wa,-adhlns="$@.lst"
```
Comment (jeanthom) [inline Makefile]: Unused

**Review by jeanthom — COMMENTED** [review]
Overall everything is good, there are little things that should be corrected here and there. I have to make the 1.4 with the current master sources before merging this release.

**Comment by phdussud** [comment]
Sorry about that. Both files should have been left out of my commit


Patrick

**Comment by jeanthom** [comment]
Changed the target branch to dirtyjtag2 branch. That created a conflict, don't worry I'll fix it.

**Comment by phdussud** [comment]
Sounds good!
Thanks for letting me know

________________________________

**Comment by phdussud** [comment]
@jeanthom  If you are interested, I did some simple work to speed up the USB traffic. I added 2 more usb buffers: I use 1 for tx and 2 for rx. The idea is that while cmd is executing we could take advantage of the USB unit to receive another packet. So as soon as we get a packet, we prepare another one for rx before calling cmd. On my machine, it cuts down the gap between jtag uploading from 200us to 50 us (at maximum speed). This gives another significant speed boost.

**Comment by jeanthom** [comment]
Sorry for radio silence, I finally got time to take care of this PR. Double buffering is indeed a good idea. Thanks again for all your contributions!

**Pull request merged**